samples/web/content/apprtc/apprtc.py

```diff
@@ -361,17 +392,28 @@ def add_client_to_room(host, room_id, client_id, is_loopback):
       error = constants.RESPONSE_DUPLICATE_CLIENT
       break
 
+    if room.room_type != room_type:
+      error = constants.RESPONSE_INVALID_ROOM
+      break
+
     if occupancy == 0:
       is_initiator = True
       room.add_client(client_id, Client(is_initiator))
       if is_loopback:
         room.add_client(constants.LOOPBACK_CLIENT_ID, Client(False))
+      if allowed_clients is not None:
+        for allowed_client in allowed_clients:
+          room.add_allowed_client(allowed_client)
     else:
       is_initiator = False
       other_client = room.get_other_client(client_id)
       messages = other_client.messages
       room.add_client(client_id, Client(is_initiator))
       other_client.clear_messages()
+      # Check if the callee was the callee intended by the caller.
+      if not room.is_client_allowed(client_id):
```

Comment:
Probably move the allowed check earlier. Otherwise we nuke messages for invalid client.